cardano-node/src/Cardano/CLI/Run.hs

```diff
@@ -397,25 +421,44 @@ withRealPBFT CLIOps{coProtocol} pcc common action = do
       action cc proto
     _ -> throwIO $ ProtocolNotSupported coProtocol
 
+txSpendUTxOByronPBFT :: CC.Genesis.Config -> SigningKey -> NonEmpty TxIn -> NonEmpty TxOut -> GenTx (ByronBlockOrEBB ByronConfig)
+txSpendUTxOByronPBFT gc sk ins outs =
+  let txattrs     = mkAttributes ()
+      tx          = UnsafeTx ins outs txattrs
+      witOfTxIn (TxInUtxo txid _) = signTxId (configProtocolMagicId gc) sk txid
+      ATxAux atx awit = mkTxAux tx . V.fromList . NE.toList $ witOfTxIn <$> ins
+  in mkByronTx $ ATxAux (reAnnotate atx) (reAnnotate awit)
+
 txSpendGenesisUTxOByronPBFT :: CC.Genesis.Config -> SigningKey -> Address -> NonEmpty TxOut -> GenTx (ByronBlockOrEBB ByronConfig)
-txSpendGenesisUTxOByronPBFT gc sk genRichAddr outs =
-  let vk         = Crypto.toVerification sk
-      txattrs    = mkAttributes ()
+txSpendGenesisUTxOByronPBFT gc sk genAddr outs =
+  let txattrs    = mkAttributes ()
       tx         = UnsafeTx (pure txIn) outs txattrs
       txIn      :: CC.UTxO.TxIn
-      txIn       = handleMissingAddr $ fst <$> Map.lookup genRichAddr initialUtxo
+      txIn       = genesisUTxOTxIn gc sk genAddr
+      cheat     :: CC.UTxO.TxWitness
+      -- Below, we have to cheat to spend a genesis UTxO entry:
+      -- sign our own TxId, not that of the input Tx.
+      -- Ledger knows.
```

Comment:
Ledger knows what?

Reply:
Ledger has a special rule to validate TXes that use genesis UTxO -- it's in on the "cheating".

Reply:
I'll clarify the comment.

Reply:
Removed that piece altogether.